active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,654 @@
+});
+```
+
+To create a new Window on a new thread in a Desktop app, create the thread first:
+
+```CS
+var thread = new Thread(() =>
+{
+    Window window = new Window();
+    window.Content = new TextBlock() { Text = "Hello" };
+    window.Activate();
+    window.Closed += (_, __) => w.DispatcherQueue.InvokeShutdown();
+
+    var syncContext = new DispatcherQueueSyncronizationContext();
+    SynchronizationContext.SetSynchronizationContext(syncContext);
+
+    this.ProcessEvents(); // Message pump
+});
+
+thread.SetApartmentState(ApartmentState.STA);
+thread.Start(); 
+```
+
+> **Spec note:** The DispatcherQueueSyncronizationContext is a new API 
+[described here](https://github.com/microsoft/microsoft-ui-xaml-specs/pull/97)
+
+## Window.ExtendsContentIntoTitleBar **[New]**
+
+Gets or sets a value that specifies whether this title bar should replace the default window title bar.
+
+Setting this property to true causes the built-in title bar to be removed. 
+
+(Spec note: this is analogous to 
+[AppWindowTitleBar.ExtendsContentIntoTitleBar](https://docs.microsoft.com/uwp/api/Windows.UI.WindowManagement.AppWindowTitleBar.ExtendsContentIntoTitleBar))
+
+Setting this property only effects rendering, it does not affect pointer (such as mouse) behavior. 
+For example, you can still use the mouse to click down in that area and drag the window around 
+the desktop. To change that behavior use the `SetTitleBar` API.
+
+Extending the content into the title bar do not impact on the Window buttons 
+(Minimize, Maximize, and Close); the buttons will be still there. 
+
+> See the SetTitleBar example below for an example of ExtendsContentIntoTitleBar
+
+(Implementation note: This will be a wrap of the CoreApplicationViewTitleBar.ExtendViewIntoTitleBar
+for UWP. In Win32 will remove the Title bar from the Non-Client area when the property is true.)
+
+## Window.SetTitleBar
+
+Makes a XAML element interact with the system as if it’s the title bar.
+
+(Spec note: this is an existing method with new behavior for Desktop)
+
+For example, a user can move the window by dragging the XAML UIElement, 
+or a user can invoke the window context menu by right-clicking it. As a consequence the app no longer
+receives pointer input when the user interacts with the target UIElement or its children using touch, mouse, or pen. 
+The UIElement does still receive keyboard input.
+
+Only one UIElement can act as Title bar, so the last set wins. To use multiple objects, 
+you need to wrap them in a container element (for example a Grid).
+
+The rectangular area occupied by the element acts as the title bar for pointer purposes,
+even if the element is blocked by another element,
+even if the element is transparent.
+
+This method is typically used with the Window's `ExtendContentIntoTitleBar` property 
+set to true in order to hide the default system title bar.
+However, even when the default system title bar is not hidden, SetTitleBar can be used to make 
```

Comment:
what is the user scenario for making any part of the app behave like the title bar? Since we want a consistent look and feel across all Windows apps, it would seem like that in the 97% use case, we don't want apps to differentiate from the default behavior. I'm sure there is a valid scenario where this is desired, and it would be helpful for the uninformed to understand when that is.

Reply:
> what is the user scenario for making any part of the app behave like the title bar? Since we want a consistent look and feel across all Windows apps, it would seem like that in the 97% use case, we don't want apps to differentiate from the default behavior. I'm sure there is a valid scenario where this is desired, and it would be helpful for the uninformed to understand when that is.
>
Pretty much any modern UWP app will extend into the Titlebar area, and will place a XAML element there so you can still grab the top of the window to move the window around.

![image](https://user-images.githubusercontent.com/7389110/101952952-673adc80-3bf1-11eb-9c49-b115f351b10b.png)
_Settings_

![image](https://user-images.githubusercontent.com/7389110/101952995-7cb00680-3bf1-11eb-8a68-70a95b5e954e.png)
_Your Phone_

![image](https://user-images.githubusercontent.com/7389110/101953037-8c2f4f80-3bf1-11eb-809a-433ce18aed36.png)
_Calculator_

![image](https://user-images.githubusercontent.com/7389110/101953103-ab2de180-3bf1-11eb-884f-d0fde346ffc7.png)
_Heutro_

![image](https://user-images.githubusercontent.com/7389110/101953138-bf71de80-3bf1-11eb-985e-d1903aa9fa1a.png)
_Windows Store_

![image](https://user-images.githubusercontent.com/7389110/101953198-dadce980-3bf1-11eb-96b3-66a1d56bd15e.png)
_Alarms and Clock_

![image](https://user-images.githubusercontent.com/7389110/101953285-ffd15c80-3bf1-11eb-9bc6-1b5f8a7fa116.png)
_Maps_

![image](https://user-images.githubusercontent.com/7389110/101953343-1a0b3a80-3bf2-11eb-936c-936494f21280.png)
_Messenger (React Native)_

Reply:
Sorry, I must not have explained myself well. I'm not suggesting that extending into the title bar doesn't have value. 

But the spec, and implementation, allows me to do something like this:

![image](https://user-images.githubusercontent.com/9649518/101961020-a94a2b00-3bbd-11eb-8c33-34056d3a1745.png)

Where the "title bar" is now in the center of my app. And, unlike the spec, I can no longer interact with the top of the Window and drag it. It doesn't matter if I set `CoreApplicationViewTitleBar.ExtendViewIntoTitleBar` or not. As soon as I call `Window.SetTitleBar` the top of the window is no longer draggable

Reply:
Apps which Extend into the titlebar, will often put items into the titlebar area, like the "What's new" button in the Your PC app.

The SetTitleBar API is used to allow the hit target of the panel placed where the Titlebar would be, to be used for dragging, snapping, and double clicking to maximise a window.  Without it, anything you place in that area would eat the taps and clicks, and it would not be possible to move the windows.

``
This method lets you specify a XAML element that interacts with the system as if it’s the app window's title bar. When you use this method, you typically set the CoreApplicationViewTitleBar.ExtendViewIntoTitleBar property to true in order to hide the default system title bar. However, even when the default system title bar is not hidden, this can be used to make additional regions in your app behave like the title bar.
``

https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.window.settitlebar?view=winrt-19041

Reply:
I don't understand/recall the motivation for the existing API to allow you to make an element in the middle of the Window be the title bar (for purposes of drag etc). A Window.TitleBar property sounds more useful to me. For the moment though we're just enabling the existing API on Desktop.